app/assets/javascripts/float-kit/addon/lib/d-menu-instance.js

```diff
@@ -0,0 +1,84 @@
+import { inject as service } from "@ember/service";
+import { action } from "@ember/object";
+import { setOwner } from "@ember/application";
+import { MENU } from "float-kit/lib/constants";
+import { guidFor } from "@ember/object/internals";
+import { tracked } from "@glimmer/tracking";
+import floatKitTriggerListeners from "float-kit/lib/trigger-listeners";
+
+@floatKitTriggerListeners
+export default class DMenuInstance {
+  @service capabilities;
+  @service menu;
+  @tracked expanded = false;
+
+  id = null;
+  trigger = null;
+
+  constructor(owner, trigger, options = {}) {
+    setOwner(this, owner);
+    options = { ...MENU.options, ...options };
+    this.id = trigger.id || guidFor(trigger);
+    this.trigger = trigger;
+    this.options = options;
+    this.setupListeners();
+  }
+
+  @action
+  onMouseMove(event) {
+    const target = event.target;
+    if (
+      event.type === "mousemove" &&
+      this.trigger.contains(target) &&
+      this.expanded
+    ) {
+      return;
+    }
+    this.onTrigger(event);
+  }
+
+  @action
+  onClick(event) {
+    if (
+      event.type === "click" &&
+      this.expanded &&
+      this.untriggers.includes("click")
+    ) {
+      this.onUntrigger(event);
+      return;
+    }
+    this.onTrigger(event);
+  }
+
+  @action
+  onMouseleave(event) {
+    if (this.untriggers.includes("hover")) {
+      this.onUntrigger(event);
+    }
+  }
+
+  @action
+  onTrigger() {
+    this.options.beforeTrigger?.(this);
+    this.show();
+  }
+
+  @action
+  onUntrigger() {
+    this.close();
+  }
+
+  @action
+  show() {
+    this.expanded = true;
+  }
+  @action
+  close() {
+    this.expanded = false;
+  }
+
```

Comment:
Can we drop these and just set the property inline instead? If you agree we have this pattern in a couple other addons as well.

Reply:
yes I can do this